c/parallel/src/jit_templates/templates/operation.h

```diff
@@ -42,8 +79,20 @@ struct stateful_user_operation
   user_operation_state<Operation.size, Operation.alignment> state;
   __device__ decltype(RetT)::Type operator()(decltype(ArgTs)::Type... args)
   {
-    return reinterpret_cast<decltype(RetT)::Type (*)(void*, decltype(ArgTs)::Type...)>(
-      Operation.operation)(&state, std::move(args)...);
+    // Note: The user provided C function (Operation.operation) must match the signature:
+    // void (void* state, void* arg1, ..., void* argN, void* result_ptr)
+    using TargetCFuncPtr = void (*)(decltype(args, void())*..., void*);
```

Comment:
```suggestion
    using TargetCFuncPtr = void (*)(void *, decltype(args, void())*..., void*);
```

pointer to the state is missing.